distribution/docker/Dockerfile

```diff
@@ -18,36 +18,15 @@
 #
 
 ARG JDK_VERSION=11
+FROM alpine as extractor
 
-# The platform is explicitly specified as x64 to build the Druid distribution.
-# This is because it's not able to build the distribution on arm64 due to dependency problem of web-console. See: https://github.com/apache/druid/issues/13012
-# Since only java jars are shipped in the final image, it's OK to build the distribution on x64.
-# Once the web-console dependency problem is resolved, we can remove the --platform directive.
-FROM --platform=linux/amd64 maven:3.8.6-jdk-11-slim as builder
+ARG VERSION=
 
-# Rebuild from source in this stage
-# This can be unset if the tarball was already built outside of Docker
-ARG BUILD_FROM_SOURCE="true"
-
-RUN export DEBIAN_FRONTEND=noninteractive \
-    && apt-get -qq update \
-    && apt-get -qq -y install --no-install-recommends python3 python3-yaml
-
-COPY . /src
 WORKDIR /src
-RUN --mount=type=cache,target=/root/.m2 if [ "$BUILD_FROM_SOURCE" = "true" ]; then \
-      mvn -B -ff -q dependency:go-offline \
-      install \
-      -Pdist,bundle-contrib-exts \
-      -Pskip-static-checks,skip-tests \
-      -Dmaven.javadoc.skip=true \
-      ; fi
+COPY ./target/apache-druid-${VERSION}-bin.tar.gz .
 
-RUN --mount=type=cache,target=/root/.m2 VERSION=$(mvn -B -q org.apache.maven.plugins:maven-help-plugin:3.2.0:evaluate \
-      -Dexpression=project.version -DforceStdout=true \
-    ) \
- && tar -zxf ./distribution/target/apache-druid-${VERSION}-bin.tar.gz -C /opt \
- && mv /opt/apache-druid-${VERSION} /opt/druid
+RUN tar -zxf /src/apache-druid-${VERSION}-bin.tar.gz -C /opt \
+ && ln -s /opt/apache-druid-${VERSION} /opt/druid
```

Comment:
there were quite a few changes upstream to the docker build (e.g. https://github.com/apache/druid/pull/11299) If you've tested the build and it work then we can do it as a follow-up but, since it looks like this was done prior to 24.x and we missed it, but this might be a good time to do it and avoid further drift since it might bite us later.

Reply:
Hey @xvrl !
I have tested the build - I will submit a follow up PR for this.